security/pkg/pki/ra/k8s_ra.go

```diff
@@ -73,6 +73,12 @@ func (r *KubernetesRA) Sign(csrPEM []byte, certOpts ca.CertOpts) ([]byte, error)
 		return nil, err
 	}
 	csrName := chiron.GenCsrName()
+	signerFromRA := r.raOpts.CaSigner
+	certSigner := certOpts.CertSigner
+	if certSigner == "" {
+		certSigner = signerFromRA
+	}
+	r.raOpts.CertSigner = certSigner
```

Comment:
This seems like its modifying the KubernetesRA options, which is intended to be static I believe. This means that one CSR will impact all future CSRs. I think that we need a local variable, not modifying our full state

Reply:
Thanks! fixed

Reply:
fixed, Thanks!